wgpu-native/src/device.rs

```diff
@@ -275,24 +275,18 @@ impl DestroyedResources<back::Backend> {
             let mut operation = None;
             let (result, ptr) = {
                 let mut buffer_guard = HUB.buffers.write();
-                let buffer = &mut buffer_guard[buffer_id];
+                let mut buffer = &mut buffer_guard[buffer_id];
                 std::mem::swap(&mut operation, &mut buffer.pending_map_operation);
                 match operation.clone().unwrap() {
                     BufferMapOperation::Read(range, ..) => {
```

Comment:
before I forget, let's make this enum variant to contain `*const u8` instead of `*mut u8`, and change the `ptr.unwrap_or(ptr::null_mut())` there accordingly on a match

Reply:
The enum does contain `*const u8` currently. The problem is that map_buffer always returns a mut ptr, and the null_mut has to match that. We could have two separate functions map_buffer_read and map_buffer_write, but it didn't seem worth it to get rid of the null_mut here?

Reply:
Is it? I don't see this PR changing the enum, and this is how it's declared - https://github.com/gfx-rs/wgpu/blob/564070b56c481cae7e7d4739f4ed1283ed9b07ed/wgpu-native/src/resource.rs#L45